### PR TITLE
Feature/dynamic firebase setup

### PR DESCRIPTION
**Pull request by mohsin2596**
Need to confirm gradle configurations before I add it for review. iOS can function fine without the pList file. Will confirm about Android and update with documentation

### CODE REVIEW

**Comment by matejthetree** [comment]
noice

**Comment by mohsin2596** [comment]
Tested firebase working without json config dependancy

**Comment by mohsin2596** [comment]
Bitrise failing because I need to add script once this branch is merged

**Review by matejthetree — APPROVED** [review]

**Comment by matejthetree** [comment]
@matheus7mm please check this

**Comment by mohsin2596** [comment]
@matejthetree @matheus7mm @MohsinN 
Suggestions are welcomed. I will be adding a md file for it later as well. Complete from my end for now. I might wanna change the environment file naming convention though will check that out tomorrow

**Review by matheus7mm — APPROVED** [review]

**Review by matheus7mm — DISMISSED** [review]

File: docs/FIREBASE_CONFIG.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+- Now make subfolders for the environments you want to create. For example if you wanted to create a dev and a staging environment then we would have two subfolders. ```[PROJECT]/environments/dev``` and ```[PROJECT]/environments/staging```
+- These subfolders should have the ***google-services.json*** and the ***GoogleService-Info.plist*** that you want to use.
+
+### Run script 
+The script right now lives in a separate package called ***env_generator*** inside our project. I propose we make this into a separate pub package later on.
+
+- cd into ```packages/env_generator/lib```
+- On your terminal run ```dart env_generator.dart```
+- The script would now create ```.env.*``` files inside your environments directory
+- A placeholder empty .env file would also be created in project root. This is needed to declare it as an asset in pubspec for production build
```
Comment (matejthetree) [inline docs/FIREBASE_CONFIG.md:19]: this is tricky,

If we want to release this as a library, I'd rather have another readme to show how to integrate with bitrise. Rather then have it created automatically.

Right now I think we can merge this. Then we extract this package as a library, and open issues there.

File: docs/FIREBASE_CONFIG.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+All the generated files are added inside of gitignore by default. 
+
+### How it works? 
+The environment generator simply extracts the useful information we would require to initialize ***Firebase Options*** and add them inside of env files to be consumed by the **dotenv** package.
+
+The initializeFirebase helper method then initializes Firebase with the provided environment. 
+
+```dart 
+await initializeFirebase(env: FirebaseEnv.dev);
```
Comment (matejthetree) [inline docs/FIREBASE_CONFIG.md:29]: is this FirebaseEnv generated?

Comment (matejthetree) [reply docs/FIREBASE_CONFIG.md:29]: can we just pass in 'string' and then we can create our own statics, inside the app for our environments.

that way the lib is project agnostic completely

**Review by matejthetree — APPROVED** [review]

File: docs/FIREBASE_CONFIG.md (hunk 2 above)
Comment (mohsin2596) [reply docs/FIREBASE_CONFIG.md:29]: The enums you see here are not coming from library ... these are inside of resident only

Comment (matejthetree) [reply docs/FIREBASE_CONFIG.md:29]: ok. Still, why use enum and no 
Class FirebaseEnv {
static String production = 'prod' 
if we end up using strings from enums?

Comment (mohsin2596) [reply docs/FIREBASE_CONFIG.md:29]: The name comes from an extension and declaration of new enums is easier. I personally preferred it but it's not a biggie for me to be stuck on. If we prefer classes instead ill make a commit to resolve it

**Pull request merged**